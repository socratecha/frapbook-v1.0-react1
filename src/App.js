import React, { Component } from 'react';

class App extends Component {
    render() {
	return (
	    <div className="App">
	      <p>FRAP applications are fun and powerful</p>
	    </div>
	);
    }
}

export default App;
