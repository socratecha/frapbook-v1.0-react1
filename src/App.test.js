import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import { getCollatzSequence, Collatz, App } from './App';

it('checks getCollatzSequence on 17 and 1', () => {
    expect(getCollatzSequence(17)).toEqual(
        [17, 52, 26, 13, 40, 20, 10, 5, 16, 8, 4, 2, 1 ]
    );
    expect(getCollatzSequence(1)).toEqual([1]);
});

it('checks getCollatzSequence on non-integer input', () => {
    expect(getCollatzSequence(1.0)).toEqual([1]);
});

it('checks getCollatzSequence contains an item', () => {
    expect(getCollatzSequence(101)).toContain(101*3 + 1);
});

it('checks getCollatzSequence on string input', () => {
    expect(() => { getCollatzSequence('1'); }).toThrow(TypeError);
});

it('checks Collatz with test renderer', () => {    
    const collatz = renderer.create(<Collatz number={ 13 } />).toJSON()
    expect(collatz).toHaveProperty(['props', 'className'], 'Collatz');
    expect(collatz).toHaveProperty(['children', '1', 'children', '2']);
    expect(collatz).toHaveProperty(['children', '0', 'type'], 'h2');
    expect(collatz.children[1].children[2].children).toEqual(["20"]);
    expect(collatz).toHaveProperty(
        ['children', '1', 'children', '4', 'children', '0'],
        '5'
    );
});

it('checks Collatz bold list style with test renderer', () => {    
    const collatz = renderer.create(<Collatz number={ 13 } />).toJSON()
    expect(collatz.children[1].children[2]).toHaveStyleRule(
        'font-weight', 'bold');
});
