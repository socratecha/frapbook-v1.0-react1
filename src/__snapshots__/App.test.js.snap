// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Collatz with test renderer 1`] = `
.c0 {
  font-weight: bold;
  font-size: 90%;
  padding: 2px;
  list-style-type: none;
}

<div
  className="Collatz"
>
  <h2>
    Collatz Sequence for 
    13
  </h2>
  <ul>
    <li
      className="c0"
    >
      13
    </li>
    <li
      className="c0"
    >
      40
    </li>
    <li
      className="c0"
    >
      20
    </li>
    <li
      className="c0"
    >
      10
    </li>
    <li
      className="c0"
    >
      5
    </li>
    <li
      className="c0"
    >
      16
    </li>
    <li
      className="c0"
    >
      8
    </li>
    <li
      className="c0"
    >
      4
    </li>
    <li
      className="c0"
    >
      2
    </li>
    <li
      className="c0"
    >
      1
    </li>
  </ul>
</div>
`;
